cmd/attest_test.go

```diff
@@ -0,0 +1,339 @@
+package cmd
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"strconv"
+	"testing"
+
+	"github.com/google/go-tpm-tools/client"
+	"github.com/google/go-tpm-tools/internal/test"
+	"github.com/google/go-tpm/tpm2"
+	"github.com/google/go-tpm/tpmutil"
+)
+
+var generateKey = map[string]func(rw io.ReadWriter) (*client.Key, error){
+	"RSA": client.GceAttestationKeyRSA,
+	"ECC": client.GceAttestationKeyECC,
+}
+
+var getIndex = map[string]uint32{
+	"RSA": client.GceAKTemplateNVIndexRSA,
+	"ECC": client.GceAKTemplateNVIndexECC,
+}
+
+func GCEAKTemplateECC() tpm2.Public {
+	return tpm2.Public{
+		Type:       tpm2.AlgECC,
+		NameAlg:    tpm2.AlgSHA256,
+		Attributes: tpm2.FlagSignerDefault,
+		ECCParameters: &tpm2.ECCParams{
+			Sign: &tpm2.SigScheme{
+				Alg:  tpm2.AlgECDSA,
+				Hash: tpm2.AlgSHA256,
+			},
+			CurveID: 3,
+			Point: tpm2.ECPoint{
+				XRaw: []byte{231, 223, 129, 82, 74, 196, 103, 67, 32, 119, 206, 163, 207, 11, 118, 76, 52, 125, 98, 2, 162, 40, 24, 237, 189, 214, 161,
+					209, 93, 215, 83, 97},
+				YRaw: []byte{215, 25, 85, 187, 225, 17, 185, 91, 60, 173, 151, 178, 136, 21, 84, 56, 131, 125, 156, 50, 219, 68, 128,
+					57, 236, 173, 217, 145, 25, 63, 112, 231},
+			},
+		},
+	}
+}
+
+func GCEAKTemplateRSA() tpm2.Public {
+	return tpm2.Public{
+		Type:       tpm2.AlgRSA,
+		NameAlg:    tpm2.AlgSHA256,
+		Attributes: tpm2.FlagSignerDefault,
+		RSAParameters: &tpm2.RSAParams{
+			Sign: &tpm2.SigScheme{
+				Alg:  tpm2.AlgRSASSA,
+				Hash: tpm2.AlgSHA256,
+			},
+			KeyBits: 2048,
+			ModulusRaw: []byte{240, 53, 205, 125, 1, 223, 48, 193, 34, 51, 180, 74, 245, 203, 239, 248, 235, 135, 119, 178, 102, 200, 62, 85, 151, 199,
```

Comment:
Why set the points and modulus?

Reply:
When I tried to fetch the GCE key template from NV data , I found that it has a points field for ECC and a modulus field for RSA algo based GCE key.
That is why I have added it in the template. 

Also if I remove this then GCEAKTemplate would be the same as AKTemplate .

Reply:
That shouldn't matter since the GCEAKTemplate you have here and AKTemplate are created under different parents. Please remove, since this makes it harder to read. The points should be set when you create the key. Setting it on input is useful when loading a key generated externally from the TPM.

Reply:
Okay , I have removed points and modulus fields from GCEKeyTemplate.